client/src/components/MyProfile.js

```diff
@@ -1,9 +1,272 @@
+  const [CityHistory, setCityHistory] = useState(props.City);
+  const [CountryHistory, setCountryHistory] = useState(props.Country);
+  const [EmailHistory, setEmailHistory] = useState(props.Email);
+
+  const editOnClick = () =>
+  {
+    setEdit(true);
+  }
+
+  const saveEditOnClick = async () =>
+  {
+    // TODO: Send the edited values to the database to actually save the edit
+    const confirmed = window.confirm("Are you sure you want to save the changes?");
+    if(confirmed){
+      try{
+        // Send updated data to the server
+        const response = await fetch('http://localhost:4000/users/updateProfile', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          _id: `${sessionStorage.getItem("id")}`,
+          user: {
+            phone: `${editedPhone}`,
+            address: `${editedAddress}`,
+            postalcode: `${editedPostalcode}`,
+            city: `${editedCity}`,
+            country: `${editedCountry}`,
+            email: `${editedEmail}`
+          }
+        })
+      });
+  
+      if (response.ok) {
+  
+        // Update the history
+      setFirstnameHistory(editedFirstname);
+      setLastnameHistory(editedLastname);
+      setPhoneHistory(editedPhone);
+      setAddressHistory(editedAddress);
+      setPostalcodeHistory(editedPostalcode);
+      setCityHistory(editedCity);
+      setCountryHistory(editedCountry);
+      setEmailHistory(editedEmail);
+  
+      // Update the actual values
+      setFirstname(editedFirstname);
+      setLastname(editedLastname);
+      setPhone(editedPhone);
+      setAddress(editedAddress);
+      setPostalcode(editedPostalcode);
+      setCity(editedCity);
+      setCountry(editedCountry);
+      setEmail(editedEmail);
+  
+      // Close the edit
+      setEdit(false);
+      }
+      else{
+        // Handle error response from server
+        console.error('Failed to update profile:', response.statusText);
+      }
+      }
+      catch(error){
+        // Handle network or other errors
+      console.error('Error updating profile:', error.message);
+      }
+    }
+  
+  }
+
+  const cancelEditOnClick = () =>
+  {
+    // Bring back the old event data
+    setFirstname(FirstnameHistory);
+    setLastname(LastnameHistory);
+    setPhone(PhoneHistory);
+    setAddress(AddressHistory);
+    setPostalcode(PostalcodeHistory);
+    setCity(CityHistory);
+    setCountry(CountryHistory);
+    setEmail(EmailHistory);
+
+    // Also reset the changes to the edited values
+    setEditedFirstname(FirstnameHistory);
+    setEditedLastname(LastnameHistory);
+    setEditedPhone(PhoneHistory);
+    setEditedAddress(AddressHistory);
+    setEditedPostalcode(PostalcodeHistory);
+    setEditedCity(CityHistory);
+    setEditedCountry(CountryHistory);
+    setEditedEmail(EmailHistory);
+
+    // Close the edit
+    setEdit(false);
+  }
+
+  const handleFirstnameChange = (event) =>
```

Comment:
Personally, I think I would pass the setters themselves to the Modal as props so you wouldn't need to have separate functions in this file🤔 Something like:
```
<EditDetailsModal
      setEditedFirstname={setEditedFirstname}
      setEditedLastname={setEditedLastname}
      .
      .
      />
```
and then in the modal you could have
```
onInput={(event) => {setEditedFirstname(event.target.value)}}
```

But it's also fine like this for me, up to you if you want to change or not :D